develop/buf-breaking.sh

```diff
@@ -61,26 +64,24 @@ check_against_commit() {
   fi
 }
 
-if [[ $PR_BASE_COMMIT ]]; then
+if [ "$PR_BASE_COMMIT" != "$COMMIT" ]; then
   # We're running in GHA, using shallow clone. Fetch some commits from the PR
   # base so we can try to find the merge base.
   color "Fetching more commits from $PR_BASE_COMMIT..."
   git fetch --no-tags --no-recurse-submodules --depth=100 origin "$PR_BASE_COMMIT"
-fi
 
-color "Cloning repo to temp dir..."
-git clone . "$tmp"
+  color "Cloning repo to temp dir..."
+  git clone . "$tmp"
 
-if [[ $PR_BASE_COMMIT ]]; then
   # First check against merge base commit:
   git -C "$tmp" fetch origin "$PR_BASE_COMMIT"
   if base=$(git -C "$tmp" merge-base HEAD "$PR_BASE_COMMIT"); then
     check_against_commit "$base" "merge base"
   else
     yellow "Can't find merge base for breaking check, checking against main only"
   fi
-fi
 
-# Next check against main:
-git -C "$tmp" fetch --depth=1 https://github.com/temporalio/temporal.git "$MAIN_BRANCH"
-check_against_commit FETCH_HEAD "main"
+  # Next check against main:
+  # git -C "$tmp" fetch --depth=1 https://github.com/temporalio/temporal.git "$MAIN_BRANCH"
+  # check_against_commit FETCH_HEAD "main"
```

Comment:
I'm commenting this for now because it's messing up with PR's that are not rebased to main branch.

Reply:
:slightly_frowning_face:   The "breaking" changes from a non-rebased PR (i.e. something in the past) to main (future) should be only that things are deleted. How about instead of disabling completely, we adjust the buf rules to not complain about deletions? i.e. turn off all the `*_NO_DELETE` rules: https://buf.build/docs/configuration/v2/buf-yaml#except-1

We could do this only for the check against main and not the check against the merge base if we use two config files, or just do it for both. We'd still catch type changes and other stuff.

Reply:
But then we would miss actual fields not being properly deleted.
I'd would say, if we really need to compare against the main branch, then we should enable the merge queue so it will run the verification using the merge branch to main.

Reply:
Yeah but fields being deleted is a smaller concern than the other rules. And I would disable them for the main check only, not the PR merge base.

One concern is that the merge base thing might not work if the PR has too many commits, where the check against main always works. I guess we can switch it to pull full history so it always finds the merge base? It'll be slower but maybe it's fine.

I'm okay with this for now since the merge base check is more important, but I would like to figure out how to do this eventually, since it's important for feature branches.

Reply:
Btw if we decide we should always fetch to the merge-base, I found this to help: https://github.com/rmacklin/fetch-through-merge-base